src/components/SearchInput.tsx

```diff
@@ -10,37 +10,43 @@ import { Icon, IconName } from '@/components/Icon';
 import { IconButton } from '@/components/IconButton';
 import { Input, InputType, type InputProps } from '@/components/Input';
 
+import { isPresent } from '@/lib/typeUtils';
+
 type ElementProps = {
   onTextChange?: (value: string) => void;
   className?: string;
 };
 
 export type SearchInputProps = ElementProps & InputProps;
 
-export const SearchInput = ({ placeholder, onTextChange, className }: SearchInputProps) => {
-  const [value, setValue] = useState('');
+export const SearchInput = ({ value, placeholder, onTextChange, className }: SearchInputProps) => {
+  const [internalValue, setInternalValue] = useState('');
```

Comment:
instead of using this setup could we instead have optional
`searchValue, setSearchValue` props. Then use the original `value, setValue` if `searchValue` and `setSearchValue` are undefined?